### ISSUE TITLE
Add custom function to calculate error after ndcube rebin

### ISSUE THREAD

**Issue by andyto1234** [sunpy/ndcube#702]
### Describe the feature

I was trying to rebin some EIS spectral data with propagate_uncertainties=True. I think because the error array is UnknownUncertainty with intensity measurement, it gave the following warning:
"UserWarning: self.uncertainty is of type UnknownUncertainty which does not support uncertainty propagation."

I got away by propagating the rebinned error myself and setting it externally. Would be great to allow custom operation on the uncertainty within NDCube rebin.

### Proposed solution

Add custom operation on the uncertainty within NDCube rebin

**Comment by Cadair** [comment]
What form do your uncertainties take? NDCube *in theory* delegates all the uncertainty propagation to Astropy's NDData.

From a quick skim of the Astropy docs, they directly support [`StdDevUncertainty`](https://docs.astropy.org/en/stable/api/astropy.nddata.StdDevUncertainty.html#astropy.nddata.StdDevUncertainty) and [`VarianceUncertainty`](https://docs.astropy.org/en/stable/api/astropy.nddata.VarianceUncertainty.html#astropy.nddata.VarianceUncertainty).

If neither of those work, you could implement your own uncertainty class, I think a subclass of [`NDUncertainty`](https://docs.astropy.org/en/stable/api/astropy.nddata.NDUncertainty.html#astropy.nddata.NDUncertainty) with a custom `propagate` method should do it?

**Comment by DanRyanIrish** [comment]
Yes, in principle that's right.  If it turns out it doesn't work as expected, we should look into that.

**Comment by DanRyanIrish** [comment]
@andyto1234 Did this resolve your issue?
